code/_onclick/click.dm

```diff
@@ -45,116 +44,110 @@
+	if(modifiers["middle"] && MiddleClickOn(target))
 		return
 	if(modifiers["shift"] && modifiers["right"])
-		ShiftRightClickOn(A)
+		INVOKE_ASYNC(src, .proc/ShiftRightClickOn, target)
 		return
-	if(modifiers["shift"] && ShiftClickOn(A))
+	if(modifiers["shift"] && ShiftClickOn(target))
 		return
 	if(modifiers["alt"] && modifiers["right"])
-		AltRightClickOn(A)
+		INVOKE_ASYNC(src, .proc/AltRightClickOn, target)
 		return
-	if(modifiers["alt"] && AltClickOn(A))
+	if(modifiers["alt"] && AltClickOn(target))
 		return
 	if(modifiers["ctrl"])
-		CtrlClickOn(A)
+		INVOKE_ASYNC(src, .proc/CtrlClickOn, target)
 		return
-	if(modifiers["right"] && RightClickOn(A))
+	if(modifiers["right"] && RightClickOn(target))
 		return
 
 	if(incapacitated(TRUE))
 		return
 
-	face_atom(A)
+	face_atom(target)
 
 	if(next_move > world.time)
 		return
 
-	if(!modifiers["catcher"] && A.IsObscured())
-		return
-
-	if(istype(loc, /obj/vehicle/multitile/root/cm_armored))
-		var/obj/vehicle/multitile/root/cm_armored/N = loc
-		N.click_action(A, src, params)
+	if(!modifiers["catcher"] && target.IsObscured())
 		return
 
 	if(restrained())
 		changeNext_move(CLICK_CD_HANDCUFFED)
-		RestrainedClickOn(A)
+		INVOKE_ASYNC(src, .proc/RestrainedClickOn, target)
 		return
 
 	if(in_throw_mode)
-		throw_item(A)
+		INVOKE_ASYNC(src, .proc/throw_item, target)
 		return
 
-	var/obj/item/W = get_active_held_item()
+	var/obj/item/held_item = get_active_held_item()
 
-	if(W == A)
-		W.attack_self(src)
+	if(held_item == target)
+		INVOKE_ASYNC(held_item, /obj/item/proc/attack_self, src)
 		update_inv_l_hand()
 		update_inv_r_hand()
 		return
 
 	//These are always reachable.
 	//User itself, current loc, and user inventory
-	if(A in DirectAccess())
-		if(W)
-			W.melee_attack_chain(src, A, params, modifiers["right"])
-		else
-			UnarmedAttack(A, FALSE, modifiers)
+	if(target in DirectAccess())
+		if(held_item)
+			INVOKE_ASYNC(held_item, /obj/item/proc/melee_attack_chain, src, target, params, modifiers["right"])
+			return
+		INVOKE_ASYNC(src, .proc/UnarmedAttack, target, FALSE, modifiers)
 		return
 
 	//Can't reach anything else in lockers or other weirdness
 	if(!loc.AllowClick())
 		return
 
-	//Standard reach turf to turf or reaching inside storage
-	if(CanReach(A, W))
-		if(W)
-			W.melee_attack_chain(src, A, params, modifiers["right"])
-		else
-			UnarmedAttack(A, TRUE, modifiers)
-	else
-		if(W)
-			var/proximity = A.Adjacent(src)
-			if(!proximity || !A.attackby(W, src, params))
-				W.afterattack(A, src, proximity, params)
-		else
-			if(A.Adjacent(src))
-				A.attack_hand(src)
-			RangedAttack(A, params)
+	if(CanReach(target, held_item))
+		if(held_item)
+			INVOKE_ASYNC(held_item, /obj/item/proc/melee_attack_chain, src, target, params, modifiers["right"])
+			return
+		INVOKE_ASYNC(src, .proc/UnarmedAttack, target, TRUE, modifiers)
+		return
+	
+	if(held_item)
+		var/proximity = target.Adjacent(src)
+		if(!proximity || !target.attackby(held_item, src, params))
```

Comment:
This lane has issue, i'm sure some attackby sleep. Probably gonna make it a signal or something